internal/xds/xdsdepmgr/xds_dependency_manager_test.go

```diff
@@ -817,63 +1213,256 @@ func (s) TestRouteResourceChangeToInline(t *testing.T) {
+	}
 	if err := mgmtServer.Update(ctx, resources); err != nil {
 		t.Fatal(err)
 	}
 
-	// Wait for the second update and verify it has the new cluster.
-	wantXdsConfig.Listener.InlineRouteConfig = &xdsresource.RouteConfigUpdate{
-		VirtualHosts: []*xdsresource.VirtualHost{
-			{
+	dm := xdsdepmgr.New(defaultTestServiceName, defaultTestServiceName, xdsClient, watcher)
+	defer dm.Close()
+
+	wantXdsConfig := &xdsresource.XDSConfig{
+		Listener: &xdsresource.ListenerUpdate{
+			RouteConfigName: defaultTestRouteConfigName,
+			HTTPFilters:     []xdsresource.HTTPFilter{{Name: "router"}},
+		},
+		RouteConfig: &xdsresource.RouteConfigUpdate{
+			VirtualHosts: []*xdsresource.VirtualHost{{
 				Domains: []string{defaultTestServiceName},
-				Routes: []*xdsresource.Route{{Prefix: newStringP("/"),
-					WeightedClusters: []xdsresource.WeightedCluster{{Name: newClusterName, Weight: 100}},
+				Routes: []*xdsresource.Route{{
+					Prefix:           newStringP("/"),
+					WeightedClusters: []xdsresource.WeightedCluster{{Name: defaultTestClusterName, Weight: 100}},
 					ActionType:       xdsresource.RouteActionRoute,
 				}},
+			}},
+		},
+		VirtualHost: &xdsresource.VirtualHost{
+			Domains: []string{defaultTestServiceName},
+			Routes: []*xdsresource.Route{{
+				Prefix:           newStringP("/"),
+				WeightedClusters: []xdsresource.WeightedCluster{{Name: defaultTestClusterName, Weight: 100}},
+				ActionType:       xdsresource.RouteActionRoute,
+			}},
+		},
+		Clusters: map[string]*xdsresource.ClusterResult{
+			defaultTestClusterName: {
+				Err: fmt.Errorf("[xDS node id: %v]: %v", nodeID, fmt.Errorf("aggregate cluster graph has no leaf clusters")),
+			},
+			clusterNameB: {
+				Err: fmt.Errorf("[xDS node id: %v]: %v", nodeID, fmt.Errorf("aggregate cluster graph has no leaf clusters")),
 			},
 		},
 	}
-	wantXdsConfig.Listener.RouteConfigName = ""
-	wantXdsConfig.RouteConfig.VirtualHosts[0].Routes[0].WeightedClusters[0].Name = newClusterName
-	wantXdsConfig.VirtualHost.Routes[0].WeightedClusters[0].Name = newClusterName
+
 	if err := verifyXDSConfig(ctx, watcher.updateCh, watcher.errorCh, wantXdsConfig); err != nil {
 		t.Fatal(err)
 	}
+}
 
-	// Change the route resource back to non-inline.
-	listener = e2e.DefaultClientListener(defaultTestServiceName, defaultTestRouteConfigName)
-	route = e2e.DefaultRouteConfig(defaultTestRouteConfigName, defaultTestServiceName, defaultTestClusterName)
-	resources = e2e.UpdateOptions{
+// Tests the case where nested aggregate clusters exceed the max depth of 16.
+// Verify that the error is correctly received in the XDSConfig in all the
+// clusters.
+func (s) TestAggregateClusterMaxDepth(t *testing.T) {
+	const clusterDepth = 17
+	nodeID, mgmtServer, xdsClient := setupManagementServerAndClient(t, true)
+
+	watcher := &testWatcher{
+		updateCh: make(chan *xdsresource.XDSConfig),
+		errorCh:  make(chan error),
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 50*defaultTestTimeout)
```

Comment:
If you are increasing it for testing, please change the value of the constant (in one place). This is very easy to miss in a review.